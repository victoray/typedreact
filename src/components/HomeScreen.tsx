import React, {useState} from "react";
import SideBar from "./SideBar";
import NavBar from "./NavBar";


const HomeScreen = () => {
    const [visible, setVisible] = useState(false);

    return (
        <div className={"container"}>
            <SideBar visible={visible}/>
            <NavBar setVisible={setVisible} visible={visible}/>
            <div className="pusher">
            </div>
        </div>
    );
};

export default HomeScreen;
