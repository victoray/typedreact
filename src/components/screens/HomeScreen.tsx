import React, {useState} from "react";
import SideBar from "../SideBar";
import NavBar from "../NavBar";
import Table from "../Table";


/**
 *
 * HomeScreen Component: This is the main screen of the app, hosting the SideBar and NavBar as child components.
 */
const HomeScreen = () => {
    const [visible, setVisible] = useState(false);

    return (
        <div className={"container"}>
            <SideBar visible={visible}/>
            <NavBar setVisible={setVisible} visible={visible}/>
            <div className="pusher">
                <div className={"ui container"} style={{marginTop: "100px"}}>
                    <Table/>
                </div>
            </div>
        </div>
    );
};

export default HomeScreen;
