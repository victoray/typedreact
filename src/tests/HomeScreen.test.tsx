import React from "react";
import HomeScreen from "../components/screens/HomeScreen";
import renderer from "react-test-renderer";
import {BrowserRouter as Router} from "react-router-dom";

const app = (
    <Router>
        <HomeScreen/>
    </Router>);

it("should render correctly", () => {
    const tree = renderer
        .create(app)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
