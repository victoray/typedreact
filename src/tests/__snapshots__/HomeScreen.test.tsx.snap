// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="ui  sidebar left fixed vertical menu"
  >
    <div
      className="item"
    >
      <img
        alt=""
        className="ui mini image"
        src="https://semantic-ui.com/images/logo.png"
      />
    </div>
    <a
      className="item"
      href="/dashboard"
      onClick={[Function]}
    >
      DashBoard 
      <i
        className="chart line icon"
      />
    </a>
    <a
      className="item"
      href="/starwars"
      onClick={[Function]}
    >
      StarWars 
      <i
        className="star icon"
      />
    </a>
    <a
      className="item"
      href="/avengers"
      onClick={[Function]}
    >
      Avengers 
      <i
        className="nintendo switch icon"
      />
    </a>
    <a
      className="item"
      href="/settings"
      onClick={[Function]}
    >
      Settings 
      <i
        className="cogs icon"
      />
    </a>
  </div>
  <div
    className="ui top fixed menu borderless"
  >
    <div
      className="link item"
    >
      <img
        alt=""
        className="ui mini image"
        src="https://semantic-ui.com/images/logo.png"
      />
    </div>
    <div
      className="link item"
      onClick={[Function]}
    >
      <i
        className="sidebar icon"
      />
      Menu
    </div>
    <div
      className="ui  category search item"
    >
      <div
        className="ui large icon input"
      >
        <input
          className="prompt"
          placeholder="Search..."
          type="text"
        />
        <i
          className="search link icon"
        />
      </div>
    </div>
    <div
      className="right menu"
    >
      <div
        className="ui simple dropdown  item"
      >
        <img
          alt=""
          className="ui avatar image"
          src="https://semantic-ui.com/images/avatar/small/matt.jpg"
        />
        <span>
          Username
        </span>
         
        <i
          className="dropdown icon"
        />
        <div
          className="menu"
        >
          <div
            className="item link"
          >
            Logout 
            <i
              className="sign-out icon"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="pusher"
  >
    <div
      className="ui container"
      style={
        Object {
          "marginTop": "100px",
        }
      }
    >
      <div
        className="ui active dimmer"
      >
        <div
          className="ui big text loader"
        >
           
          Fetching Data
        </div>
      </div>
    </div>
  </div>
</div>
`;
